packages/backend/server/src/plugins/copilot/providers/anthropic.ts

```diff
@@ -146,26 +159,76 @@ export class AnthropicProvider
     try {
       metrics.ai.counter('chat_text_stream_calls').add(1, { model });
       const [system, msgs] = await chatToGPTMessage(messages);
-
-      const { textStream } = streamText({
+      const { fullStream } = streamText({
         model: this.#instance(model),
         system,
         messages: msgs,
         abortSignal: options.signal,
+        providerOptions: {
+          anthropic: this.getAnthropicOptions(options),
+        },
+        tools: {
+          webSearch: createExaTool(this.AFFiNEConfig),
+        },
+        maxSteps: this.MAX_STEPS,
+        experimental_continueSteps: true,
       });
 
-      for await (const message of textStream) {
-        if (message) {
-          yield message;
-          if (options.signal?.aborted) {
-            await textStream.cancel();
+      for await (const message of fullStream) {
+        switch (message.type) {
+          case 'reasoning': {
+            yield message.textDelta;
+            break;
+          }
```

Comment:
how does the frontend distinguish between reasoning and normal messages?

Reply:
Wait for markdown callout support